app/models/name/merge.rb

```diff
@@ -130,4 +134,40 @@ def merge(old_name)
     # Finally destroy the name.
     old_name.destroy
   end
+
+  ##############################################################################
+
+  private
+
+  def approved_synonym_of_proposed_name?
+    !deprecated && (other_synonym_ids & Naming.all.pluck(:name_id)).present?
+  end
+
+  def correctly_spelled_ancestor_of_proposed_name?
+    return false if correct_spelling.present?
+    return above_genus_is_ancestor? if classified_above_genus?
+    return genus_or_species_is_ancestor? if [:Genus, :Species].include?(rank)
+    return group_is_ancestor? if rank == :Group
+
+    false
+  end
+
+  def above_genus_is_ancestor?
+    Name.joins(:namings).where(
+      "classification LIKE ?", "%#{rank}: #{classification_name}%"
```

Comment:
Sorry to be picky, but is this the only place classification_name is used?  All it does is add underscores before and after the name, which can be done here in-line without any risk of increasing ABC complexity or anything.  Then we would not have to look up classification_name to see whether it is using text_name or search_name or something else under the hood.  I really wish the number of levels of method calls within method calls within method calls within... was considered in the calculations of what is considered "complexity".  Trivial helper methods like this contribute non-trivially to the complexity and difficulty of understanding code, in my opinion.  Now, if you were doing names.sort_by(&:classification_name) or names.map(&:classification_name), that would be a different matter.  But in this particular instance I don't see that it buys us much of anything.  Oh... but if we also used it in the classification code, that would also be a different matter.  That would enable us to change the format in which names are stored in the classification string throughout the site in one place.  But it doesn't look like we're actually doing that.

Reply:
I agree. It's used only once, and I think the intent is sufficiently clear without a help method.
I'll use string interpolation instead.